### PR TITLE
add datadog apm

### PR DESCRIPTION
**Pull request by jonnymccullagh**
## Description
Adding support for datadog APM tracing:
- Helm chart changes for annotations and labels in the relevant parts of deployment file
- Added dd-trace npm package to package.json for server and worker
- Init the datadog tracer in the worker and app service

## Testing
- I do not have the necessary skills to test the node tracing so would appreciate it if someone else could give that a shot OR I can test in QA

Related: 
- https://github.com/Budibase/budibase-deploys/pull/23

### CODE REVIEW

File: charts/budibase/templates/app-service-deployment.yaml [hunk 1]
```diff
@@ -157,6 +166,10 @@ spec:
         - name: NODE_DEBUG
           value: {{ .Values.services.apps.nodeDebug | quote }}
         {{ end }}
+        {{ if .Values.globals.datadogApmEnabled }}
+        - name: DD_LOGS_INJECTION
```
[the remark this answers is not in the document]
Comment (Rory-Powell) [reply charts/budibase/templates/app-service-deployment.yaml:170]: There's also a built in var here

https://docs.datadoghq.com/tracing/trace_collection/library_config/nodejs/

For "DD_TRACE_ENABLED". 

I think we should also add the "DD_ENV" aliased to the existing budibase environment value

File: packages/server/src/app.ts [hunk 2]
```diff
@@ -1,6 +1,11 @@
 // need to load environment first
 import env from "./environment"
 
+// enable APM if configured
+if (process.env.DATADOG_APM_ENABLED) {
+  const tracer = require("dd-trace").start()
```
[the remark this answers is not in the document]
Comment (jonnymccullagh) [reply packages/server/src/app.ts]: Yes, fixed.

**Comment by jonnymccullagh** [comment]
I have read the CLA Document and I hereby sign the CLA

File: charts/budibase/templates/app-service-deployment.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rory-Powell) [reply charts/budibase/templates/app-service-deployment.yaml:170]: Ah I see it's all done using labels instead, cool beans

[the remark this answers is not in the document]
Comment (Rory-Powell) [reply charts/budibase/templates/app-service-deployment.yaml:170]: I think the original point is still true about the DATADOG_APM_ENABLED, our application will rely on that to start the tracer

[the remark this answers is not in the document]
Comment (jonnymccullagh) [reply charts/budibase/templates/app-service-deployment.yaml:170]: ok adding it now

[the remark this answers is not in the document]
Comment (jonnymccullagh) [reply charts/budibase/templates/app-service-deployment.yaml:170]: DD_APM_ENABLED default value is true according to docs but I have added it along with the DD_URL pointing to the eu domain name - belt and braces.

**Comment by jonnymccullagh** [comment]
I have read the CLA Document and I hereby sign the CLA

File: charts/budibase/templates/app-service-deployment.yaml [hunk 3]
```diff
@@ -4,28 +4,37 @@ metadata:
...
+{{ if .Values.services.apps.deploymentLabels }}
+{{- toYaml .Values.services.apps.deploymentLabels | indent 4 -}}
+{{ end }}
   name: app-service
 spec:
   replicas: {{ .Values.services.apps.replicaCount }}
   selector:
     matchLabels:
-      io.kompose.service: app-service
+      io.kompose.service: app-service
```
Comment (Rory-Powell) [inline charts/budibase/templates/app-service-deployment.yaml:21]: shouldn't be a problem right? maybe be safe to remove the extra space

**Review by Rory-Powell — APPROVED** [review]

**Pull request merged**